Tests/Leb128.lean

```diff
@@ -0,0 +1,138 @@
+def bitlebx : List Bit := ByteArray.toBits blebx
+def ulebx : List Bit := unlead bitlebx
+def plebx : List Bit := pad7 ulebx
+def nplebx : List Bit := npad7 lebx
+
+/-
+5> erlang:length( "000010011000011101100101" ).
+24
+8> erlang:length("00001001").
+8
+9> erlang:length("10000111").
+8
+10> erlang:length("01100101").
+8
+11> [ 2#00001001, 2#10000111, 2#01100101 ].
+-/
+def testBlebxIs9_135_101 : TestSeq := Id.run $ do
+  let expected : ByteArray := ByteArray.mk #[9,135,101]
+  let actual := ntob lebx
+  test "Big endian 624485 should be ByteArray.mk #[9,135,101]" $
+    expected.data == actual.data &&
+    actual.data == blebx.data
+
+def testBitlebxIs000010011000011101100101 : TestSeq := Id.run $ do
+  -- 00001 00110 00011 10110 0101 with spaces between each bit:
+  --                          0 0 0 0 1|0 0 1 1 0|0 0 0 1 1|1 0 1 1 0|0 1 0 1
+  let expected : List Bit := [0,0,0,0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]
+  let actual := ByteArray.toBits blebx
+  --                      0 0 0 0 1|0 0 1 1 0|0 0 0 1 1|1 0 1 1 0|0 1 0 1
+  test "Big endian 624485 converted to bits should be [0,0,0,0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]" $
+    expected == actual &&
+    actual == bitlebx
+
+def testUlebxIs10011000011101100101 : TestSeq := Id.run $ do
+  -- 10011 00001 11011 00101 with spaces between each bit:
+  --                          1 0 0 1 1|0 0 0 0 1|1 1 0 1 1|0 0 1 0 1
+  -- ===                      1|0 0 1 1 0|0 0 0 1 1|1 0 1 1 0|0 1 0 1
+  let expected : List Bit := [1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]
+  let actual := unlead bitlebx
+  test "Big endian 624485 in binary without leading zeroes should be [1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]" $
+    expected == actual &&
+    actual == ulebx
+
+def testPad7UlebxIs010011000011101100101 : TestSeq := Id.run $ do
+  -- 01001 10000 11101 10010 1 with spaces between each bit:
+  --                          0 1 0 0 1|1 0 0 0 0|1 1 1 0 1|1 0 0 1 0|1
+  let expected : List Bit := [0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]
+  let actual := pad7 ulebx
+  test "Big endian 624485 in binary without leading zeroes 7-bit padded should be [0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]" $
+    expected == actual
+
+def testCanPad7Naturals : TestSeq := Id.run $ do
+  -- 01001 10000 11101 10010 1 with spaces between each bit:
+  --                          0 1 0 0 1|1 0 0 0 0|1 1 1 0 1|1 0 0 1 0|1
+  let expected : List Bit := [0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]
+  let actual := npad7 lebx
+  test "Number 624485, 7-bit padded should be [0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]" $
+    expected == actual &&
+    actual == nplebx &&
+    actual == plebx
+
+def testUnsignedLeb128 : TestSeq := Id.run $ do
+  let expected : ByteArray := ByteArray.mk #[229, 142, 38]
+  let actual := uLeb128 lebx
+  test "Unsigned LEB128 of 624485 should be ByteArray.mk #[ 229, 142, 38 ] and 1 is ByteArray.mk #[1]" $
+    expected.data == actual.data &&
+    (uLeb128 1).data == (ByteArray.mk #[1]).data
```

Comment:
Does `DecidableEq` (i.e. `=`) not work on `ByteArray`s? Why `.data`? To put it in to stdlib was my first Lean4 bug report 😅

Reply:
Nothing else compiled. I tried `=`, I tried without `data`, I don't remember error messages, sorry.

I kinda don't care to fix this because the test sizes are very small.